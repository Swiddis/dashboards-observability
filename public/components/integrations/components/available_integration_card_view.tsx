/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton, EuiCard, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
import _ from 'lodash';
import React from 'react';
import {
  AvailableIntegrationsCardViewProps,
  AvailableIntegrationType,
} from './available_integration_overview_page';
import { INTEGRATIONS_BASE } from '../../../../common/constants/shared';

export function AvailableIntegrationsCardView(props: AvailableIntegrationsCardViewProps) {
  const getImage = (url?: string) => {
    let optionalImg;
    if (url) {
      optionalImg = (
        <img style={{ height: 100, width: 100 }} alt="" className="synopsisIcon" src={url} />
      );
    }
    return optionalImg;
  };

  const renderRows = (integrations: AvailableIntegrationType[]) => {
    if (!integrations || !integrations.length) return null;
    return (
      <>
        <EuiFlexGroup gutterSize="l" style={{ flexWrap: 'wrap' }}>
          {integrations.map((i, v) => {
            return (
              <EuiFlexItem key={v} style={{ minWidth: '14rem', maxWidth: '14rem' }}>
                <EuiCard
                  icon={getImage(
                    `${INTEGRATIONS_BASE}/repository/${i.name}/static/${i.statics.logo.path}`
                  )}
                  title={i.name}
                  description={i.description}
                  data-test-subj={`integration_card_${i.name.toLowerCase()}`}
                  titleElement="span"
                  onClick={() => (window.location.hash = `#/available/${i.name}`)}
                />
              </EuiFlexItem>
            );
          })}
        </EuiFlexGroup>
        <EuiSpacer />
      </>
    );
  };

  return <>{renderRows(props.data.hits)}</>;
}
