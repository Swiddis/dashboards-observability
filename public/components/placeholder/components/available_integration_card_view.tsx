import {
  EuiButton,
  EuiCard,
  EuiFlexGrid,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHorizontalRule,
  EuiIcon,
  EuiPage,
  EuiPageBody,
  EuiPageContent,
  EuiPageHeader,
  EuiPageHeaderSection,
  EuiPanel,
  EuiSelectOption,
  EuiSpacer,
  EuiTabbedContent,
  EuiTabbedContentTab,
  EuiText,
  EuiTitle,
} from '@elastic/eui';
import {
  OuiButton,
  OuiCard,
  OuiIcon,
  OuiFlexGroup,
  OuiFlexItem,
  OuiLink,
  OuiSpacer,
  OuiText,
} from '@opensearch-project/oui';
import _ from 'lodash';
import DSLService from 'public/services/requests/dsl';
import PPLService from 'public/services/requests/ppl';
import SavedObjects from 'public/services/saved_objects/event_analytics/saved_objects';
import TimestampUtils from 'public/services/timestamp/timestamp';
import React, { ReactChild, useEffect, useState } from 'react';
import {
  AvailableIntegrationsCardViewProps,
  AvailableIntegrationType,
} from './available_integration_overview_page';

export function AvailableIntegrationsCardView(props: AvailableIntegrationsCardViewProps) {
  const rowNumber = props.records / 5;
  //   console.log(rowNumber)

  //             title={feature}
  //             onClick={() => {
  //               window.location.assign(`#/placeholder/${feature}`);
  //             }}

  const getImage = (url?: string) => {
    let optionalImg;
    if (url) {
      optionalImg = <img alt="" className="synopsisIcon" src={url} />;
    }
    return optionalImg;
  };

  // const classes = classNames('homSynopsis__card', {
  //   'homSynopsis__card--noPanel': !wrapInPanel,
  // });

  const renderRows = (integrations: AvailableIntegrationType[]) => {
    return _.times(rowNumber).map(() => {
      return (
        <>
          <EuiFlexGroup gutterSize="l">
            {integrations.map((i, v) => {
              return (
                <EuiFlexItem key={v}>
                  <EuiCard
                    // className={classes}
                    layout="vertical"
                    icon={getImage(i.assetUrl)}
                    titleSize="xs"
                    title={i.templateName}
                    description={i.description}
                    data-test-subj={`homeSynopsisLink${i.templateName.toLowerCase()}`}
                    footer={
                      <div>
                        <EuiButton
                          aria-label="Go to Developers Tools"
                          onClick={() => {
                            window.location.assign(`#/placeholder/${i.templateName}`);
                          }}
                        >
                          View Details
                        </EuiButton>
                        <EuiSpacer />
                        <EuiButton aria-label="Go to Developers Tools" onClick={() => {}}>
                          Add
                        </EuiButton>
                      </div>
                    }
                  />
                </EuiFlexItem>
              );
            })}
          </EuiFlexGroup>
          <EuiSpacer />
        </>
      );
    });
  };

  return <>{renderRows(props.data.data)}</>;
}

//   Synopsis.propTypes = {
//     description: PropTypes.string.isRequired,
//     iconUrl: PropTypes.string,
//     iconType: PropTypes.string,
//     title: PropTypes.string.isRequired,
//     url: PropTypes.string,
//     onClick: PropTypes.func,
//     isBeta: PropTypes.bool,
//   };

//   Synopsis.defaultProps = {
//     isBeta: false,
//   };
