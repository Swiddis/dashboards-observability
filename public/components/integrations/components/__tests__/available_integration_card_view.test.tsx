/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { waitFor } from '@testing-library/react';
import { AvailableIntegrationsCardView } from '../available_integration_card_view';
import { data } from './testing_constants';

describe('Integration Header Test', () => {
  configure({ adapter: new Adapter() });

  it('Renders integration header as expected', async () => {
    const wrapper = mount(AvailableIntegrationsCardView(data));

    await waitFor(() => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
