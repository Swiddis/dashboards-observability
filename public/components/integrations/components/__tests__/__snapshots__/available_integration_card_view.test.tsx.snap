// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Available Integration Card View Test Renders nginx integration card view using dummy data 1`] = `
Array [
  <EuiFlexGroup
    gutterSize="l"
    style={
      Object {
        "flexWrap": "wrap",
      }
    }
  >
    <div
      className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      style={
        Object {
          "flexWrap": "wrap",
        }
      }
    >
      <EuiFlexItem
        key="0"
        style={
          Object {
            "maxWidth": "14rem",
            "minWidth": "14rem",
          }
        }
      >
        <div
          className="euiFlexItem"
          style={
            Object {
              "maxWidth": "14rem",
              "minWidth": "14rem",
            }
          }
        >
          <EuiCard
            data-test-subj="integration_card_nginx"
            description="Nginx HTTP server collector"
            icon={
              <img
                alt=""
                className="synopsisIcon"
                src="/api/integrations/repository/nginx/static/logo"
                style={
                  Object {
                    "height": 100,
                    "width": 100,
                  }
                }
              />
            }
            onClick={[Function]}
            title="nginx"
            titleElement="span"
          >
            <EuiPanel
              className="euiCard euiCard--centerAligned euiCard--isClickable euiCard--hasIcon"
              data-test-subj="integration_card_nginx"
              element="div"
              hasShadow={true}
              onClick={[Function]}
            >
              <div
                className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--isClickable euiCard euiCard--centerAligned euiCard--isClickable euiCard--hasIcon"
                data-test-subj="integration_card_nginx"
                onClick={[Function]}
              >
                <div
                  className="euiCard__top"
                >
                  <img
                    alt=""
                    className="synopsisIcon euiCard__icon"
                    src="/api/integrations/repository/nginx/static/logo"
                    style={
                      Object {
                        "height": 100,
                        "width": 100,
                      }
                    }
                  />
                </div>
                <div
                  className="euiCard__content"
                >
                  <EuiTitle
                    className="euiCard__title"
                    id="random_html_idTitle"
                    size="s"
                  >
                    <span
                      className="euiTitle euiTitle--small euiCard__title"
                      id="random_html_idTitle"
                    >
                      <button
                        aria-describedby=" random_html_idDescription"
                        className="euiCard__titleButton"
                        disabled={false}
                        onClick={[Function]}
                      >
                        nginx
                      </button>
                    </span>
                  </EuiTitle>
                  <EuiText
                    className="euiCard__description"
                    id="random_html_idDescription"
                    size="s"
                  >
                    <div
                      className="euiText euiText--small euiCard__description"
                      id="random_html_idDescription"
                    >
                      <p>
                        Nginx HTTP server collector
                      </p>
                    </div>
                  </EuiText>
                </div>
              </div>
            </EuiPanel>
          </EuiCard>
        </div>
      </EuiFlexItem>
    </div>
  </EuiFlexGroup>,
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>,
]
`;
